### ISSUE TITLE
[Bug]: Published beta/edge is older then stable/candidate

### ISSUE THREAD

**Issue by smarek** [snapcrafters/signal-desktop#92]
### What happened?

1. periodic "snap refresh" downgraded my signal (installed via "snap refresh signal-desktop --edge") to version older than which was already installed
2. signal-desktop offered to delete all my local data and start over, this could be potentialy hazardous to users data (user will accidentaly remove everything, possibly without backup or linked device, that would hold the history of conversations)

### What should have happened?

1. published versions should be higher, the more unstable channel you use
2. downgrade should not happen without confirmation

### Output of `snap info $snap_name`

```shell
name:      signal-desktop
summary:   Private messaging from your desktop.
publisher: Snapcrafters
store-url: https://snapcraft.io/signal-desktop
contact:   https://github.com/snapcrafters/signal-desktop/issues
license:   AGPL-3.0-only
description: |
  Private messaging from your desktop.
  
  To use the Signal desktop app, Signal must first be installed on your phone.
  
  **Are you having issues?**
  
  Let us know by creating a new issue here: https://github.com/snapcrafters/signal-desktop/issues
  
  **Authors**
  
  This snap is maintained by the Snapcrafters community, and is not necessarily endorsed or
  officially maintained by the upstream developers.
commands:
  - signal-desktop
snap-id:      r4LxMVp7zWramXsJQAKdamxy6TAWlaDD
tracking:     latest/stable
refresh-date: today at 17:43 CET
channels:
  latest/stable:    6.2.0  2023-01-14 (428) 188MB -
  latest/candidate: 6.2.0  2023-01-14 (428) 188MB -
  latest/beta:      5.62.0 2022-11-24 (400) 192MB -
  latest/edge:      ↑                             
installed:          6.2.0             (428) 188MB -
```

### Output of `snap connections $snap_name`

```shell
irrelevant
```

### Output of `snap version`

```shell
snap    2.58
snapd   2.58
series  16
debian  -
kernel  6.0.0-6-amd64
```

### Relevant log output

```shell
MainSQL: Database startup error: Error: SQL: User version is 71 but the expected maximum version is 66. Did you try to start an old version of Signal?\n    at updateSchema ([REDACTED]/ts/sql/migrations/index.js:1692:11)\n    at Object.initialize ([REDACTED]/ts/sql/Server.js:376:40)\n    at MessagePort.<anonymous> ([REDACTED]/ts/sql/mainWorker.js:68:35)\n    at [nodejs.internal.kHybridDispatch] (node:internal/event_target:643:20)\n    at exports.emitMessage (node:internal/per_context/messageport:23:28)
```

### Teminal output of app

```shell
terminal output as whole is irrelevant, important snippet extracted
```

**Comment by merlijn-sebrechts** [comment]
I agree this was less than ideal. I apologize for the disruption. This happened because we switched from build systems.

* We used to build using the snapcraft build service which published to `edge`.
* Now we switched to GitHub, which publishes to `candidate`.
* The older version in the `beta` channel is there for people who use Centos, because the latest version doesn't work on Centos anymore.

I closed the `edge` channel, because it's not used anymore, but this kicked the people still using `edge` to the `beta` channel instead of to the `candidate` channel. The good news is that this won't happen again, now that the `edge` channel is closed for good.
